libcst/metadata/scope_provider.py

```diff
@@ -86,6 +99,84 @@ class BuiltinAssignment(BaseAssignment):
     pass
 
 
+class QualifiedNameSource(Enum):
+    IMPORT = auto()
+    LOCAL = auto()
+
+
+@dataclass(frozen=True)
+class QualifiedName:
+    name: str
+    source: QualifiedNameSource
+
+
+class _QualifiedNameUtil:
+    @staticmethod
+    def get_full_name_for(node: cst.CSTNode) -> Optional[str]:
+        if isinstance(node, cst.Name):
+            return node.value
+        elif isinstance(node, cst.Attribute):
+            return (
+                f"{_QualifiedNameUtil.get_full_name_for(node.value)}.{node.attr.value}"
+            )
+        elif isinstance(node, cst.Call):
+            return _QualifiedNameUtil.get_full_name_for(node.func)
+        elif isinstance(node, cst.Subscript):
+            return _QualifiedNameUtil.get_full_name_for(node.value)
+        elif isinstance(node, cst.SimpleString):
+            # In the case of SimpleString of type hints.
+            return literal_eval(node.value)
+        return None
+
+    @staticmethod
+    def find_qualified_name_for_import_alike(
+        assignment_node: Union[cst.Import, cst.ImportFrom],
+        name_parts: List[str],
+        results: Set[QualifiedName],
+    ) -> None:
+        module = ""
+        if isinstance(assignment_node, cst.ImportFrom):
+            module_attr = assignment_node.module
+            if module_attr:
+                module = _QualifiedNameUtil.get_full_name_for(module_attr)
```

Comment:
If a module is relative-only import, it looks like we currently handle this as setting the module to "". I think that's fine, but we should probably have a TODO.

Reply:
Added TODO.